lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Context/ContextRuntime.cs

```diff
@@ -93,14 +90,35 @@ internal sealed class ContextRuntime
         }
 
         /// <summary>
+        /// Create a new ContextRuntime.
+        /// </summary>
+        /// <param name="serviceInjector"></param>
+        /// <param name="contextConfiguration">the Configuration for this context.</param>
+        /// <param name="parentContext"></param>
+        public ContextRuntime(
+                IInjector serviceInjector,
+                IConfiguration contextConfiguration,
+                Optional<ContextRuntime> parentContext)
+        {
+            _serviceInjector = serviceInjector;
+            _contextInjector = serviceInjector.ForkInjector(contextConfiguration);
+            _contextLifeCycle = _contextInjector.GetInstance<ContextLifeCycle>();
+            _parentContext = parentContext;
+            _contextLifeCycle.Start();
+        }
+
+        /// <summary>
         ///  Create a new ContextRuntime for the root context.
         /// </summary>
-        /// <param name="serviceInjector"> </param> the serviceInjector to be used.
+        /// <param name="id">the ID of the context.</param>
+        /// <param name="serviceInjector">the serviceInjector to be used.</param>
         /// <param name="contextConfiguration"> the Configuration for this context.</param>
+        [Obsolete("Deprecated in 0.14, will be removed.")]
         public ContextRuntime(
+            string id,
             IInjector serviceInjector,
             IConfiguration contextConfiguration)
-            : this(serviceInjector, contextConfiguration, Optional<ContextRuntime>.Empty())
+            : this(id, serviceInjector, contextConfiguration, Optional<ContextRuntime>.Empty())
```

Comment:
If it is going to be obsolete, why brother to change it to add id parameter here?